submission/hammer/main.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Hammer tool sends multiple add-pre-chain requests to Submission proxy at the same time.
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/schedule"
+	"github.com/google/certificate-transparency-go/submission"
+)
+
+// Default number of submissions is intentionally low.
+// After several runs, likely to hit Log rate-limits.
+var (
+	proxyEndpoint = flag.String("proxy_endpoint", "http://localhost:5951/", "Endpoint for HTTP (host:port)")
+	timeout       = flag.Duration("duration", 0*time.Minute, "Time to run continuous flow of submissions. "+
+		"When this and count-flag both have non-zero values, submission ends upon reaching earliest restriction")
+	count = flag.Int("count", 10, "Total number of submissions to execute. "+
+		"When this and timeout-flag both have non-zero values, submission ends upon reaching earliest restriction")
+	qps = flag.Int("qps", 5, "Number of requests per second")
+)
+
+func main() {
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Sorry, I should have spotted this before - doesn't `ioutil.ReadFile()` find files relative to the working directory automatically? Therefore, you shouldn't need to join with `wd`.

Reply:
Changed, thanks